src/libs/actions/Report.js

```diff
@@ -2081,7 +2113,10 @@ function leaveRoom(reportID, isWorkspaceMemberLeavingWorkspaceRoom = false) {
         },
     );
 
-    if (isWorkspaceMemberLeavingWorkspaceRoom) {
+    const lastAccessedReportID = updateRecentlyAccessedReportID(reportID, true);
+    if (lastAccessedReportID) {
+        Navigation.dismissModal(lastAccessedReportID);
```

Comment:
@getusha Should we handle the case of leave thread as well? If so, then we can change this line to
```js
Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(lastAccessedReportID));
```
Because leave thread button is in the three dot popover.

And also need to change the logic to navigate to concierge page of `ReportScreen` here
https://github.com/Expensify/App/blob/f285c9cf3294697e1cb1f9c797641876b38d5829/src/pages/home/ReportScreen.js#L302-L324

cc @techievivek

Reply:
> Should we handle the case of leave thread as well? If so, then we can change this line to

What is the behavior with the current change? IIRC, we used to drop the user to the parent report in case of a thread. Don't we have the same behavior with the new changes?

Reply:
@techievivek The current new changes can't drop user to previous chat when leaving a thread for following reasons
1. The leaving thread button is from a popover and `Navigation.dismissModal` is used dismiss RHP modal. So, we need to change use `Navigation.navigate` if take leaving thread into consideration.
2. `ReportScreen` has [logic](https://github.com/Expensify/App/blob/f285c9cf3294697e1cb1f9c797641876b38d5829/src/pages/home/ReportScreen.js#L302-L324) to redirect to parent reportID

The current changes works as follows 

https://github.com/Expensify/App/assets/117511920/1ff38227-8322-48ba-9bcd-067c7bbae750

Reply:
> The current new changes can't drop user to previous chat when leaving a thread for following reasons

We don't want to drop them to the previous chat in case of a thread report or any other report with the parent report/reportAction ID. We should just take them to the parent report.